mutable_tree.go

```diff
@@ -733,6 +733,13 @@ func (tree *MutableTree) SaveVersion() ([]byte, int64, error) {
 	}
 
 	tree.logger.Debug("SAVE TREE", "version", version)
+	tree.ndb.resetLatestVersion(version)
```

Comment:
If we move nodeDb version before saving the nodes might this expose a different kind of race condition?

Reply:
saveFastNodes requires the updated version, let me refactor it